target/linux/realtek/files-6.12/drivers/net/pcs/pcs-rtl-otto.c

```diff
@@ -666,30 +666,25 @@ static int rtpcs_930x_sds_clock_wait(struct rtpcs_ctrl *ctrl, int timeout)
 	return 1;
 }
 
-static void rtpcs_930x_sds_mac_link_config(struct rtpcs_ctrl *ctrl, int sds,
-					   bool tx_normal, bool rx_normal)
+static void rtpcs_930x_sds_set_polarity(struct rtpcs_ctrl *ctrl, int sds,
+					bool tx_inv, bool rx_inv)
 {
 	u32 v10, v1;
 
 	v10 = rtpcs_sds_read(ctrl, sds, 6, 2); /* 10GBit, page 6, reg 2 */
 	v1 = rtpcs_sds_read(ctrl, sds, 0, 0); /* 1GBit, page 0, reg 0 */
```

Comment:
Might as well use `rtpcs_sds_write_bits` instead of `rtpcs_sds_read` and then `rtpcs_sds_write` later on.
`rtpcs_930x_sds_set_polarity` literally does exactly the same as your `rtpcs_931x_sds_set_polarity` just in a different order (first rx for 10GR and 1G, then tx for 10GR und 1G). the 931x version has better readability so let's use that one instead, no?

Reply:
jep, that's what I was hinting at. they are doing exactly the same thing ^^
Does it depend on the function how (read/write) errors would need to be handled or is it always the same?
Would errors always return the function or rather retry after some time?

Upstream uses macros in certain cases:
https://github.com/torvalds/linux/blob/284922f4c563aa3a8558a00f2a05722133237fe8/include/net/dsa.h#L582
Something similar could be done here:
`exec_or_err` could be a name for a macro and the actual function would be called in it's scope. it's comparable to with statements for files in python or the dsa case above.
Why should upstream not accept this if it keeps boilerplate visibly cleaner?